server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -110,6 +110,8 @@ private static void internalParseDocument(RootObjectMapper root, MetadataFieldMa
             parseObjectOrNested(context, root);
         }
 
+        root.postParse(context, new IndexTimeScriptParams(context));
```

Comment:
Most of the time this does nothing - I think hotspot will just remove it entirely in most cases but we will want to do some perf testing against indexes with large mappings to make sure that it doesn't cause a regression.  We can always add a check that a post-parse script is present somewhere in the mappings and skip if it isn't set.